cpp/tests/utilities/column_wrapper.hpp

```diff
@@ -0,0 +1,455 @@
+ * element in `[begin,end)` that evaluated to `true`.
+ *---------------------------------------------------------------------------**/
+template <typename ValidityIterator>
+std::vector<bitmask_type> make_null_mask_vector(ValidityIterator begin,
+                                                ValidityIterator end) {
+  cudf::size_type size = std::distance(begin, end);
+  auto num_words =
+      cudf::bitmask_allocation_size_bytes(size) / sizeof(bitmask_type);
+  std::vector<bitmask_type> null_mask(num_words, 0);
+  for (auto i = 0; i < size; ++i) {
+    if (begin[i] == true) {
+      set_bit_unsafe(null_mask.data(), i);
+    }
+  }
+  return null_mask;
+}
+
+/**---------------------------------------------------------------------------*
+ * @brief Create a `device_buffer` containing a validity indicator bitmask using
+ * the range `[begin,end)` interpretted as booleans to indicate the state of
+ *each bit.
+ *
+ * If `*(begin + i) == true`, then bit `i` is set to 1, else it is zero.
+ *
+ * @tparam ValidityIterator
+ * @param begin The beginning of the validity indicator sequence
+ * @param end The end of the validity indicator sequence
+ * @return rmm::device_buffer Contains a bitmask where bits are set for every
+ * element in `[begin,end)` that evaluated to `true`.
+ *---------------------------------------------------------------------------**/
+template <typename ValidityIterator>
+rmm::device_buffer make_null_mask(ValidityIterator begin,
+                                  ValidityIterator end) {
+  auto null_mask = make_null_mask_vector(begin, end);
+  return rmm::device_buffer{
+      null_mask.data(), null_mask.size() * sizeof(decltype(null_mask.front()))};
+}
+
+/**---------------------------------------------------------------------------*
+ * @brief Given the range `[begin,end)`, converts each value to a string and
+ * then creates a packed vector of characters for each string and a vector of
+ * offsets indicating the starting position of each string.
+ *
+ * @tparam StringsIterator A `std::string` must be constructible from
+ * dereferencing a `StringsIterator`.
+ * @tparam ValidityIterator Dereferencing a ValidityIterator must be
+ * convertible to `bool`
+ * @param begin The beginning of the sequence of values to convert to strings
+ * @param end The end of the sequence of values to convert to strings
+ * @param v The beginning of the validity indicator sequence
+ * @return std::pair containing the vector of chars and offsets
+ *---------------------------------------------------------------------------**/
+template <typename StringsIterator, typename ValidityIterator>
+auto make_chars_and_offsets(StringsIterator begin, StringsIterator end,
+                            ValidityIterator v) {
+  std::vector<char> chars{};
+  std::vector<int32_t> offsets(1, 0);
+  for (; begin < end; ++begin) {
+    std::string tmp = (*v++) ? std::string(*begin) : std::string{};
```

Comment:
Slightly more intuitive loop?

```suggestion
  for (auto str = begin; str < end; ++str) {
    std::string tmp = (*v++) ? std::string(*str) : std::string{};
```